test/commands.test.ts

```diff
@@ -108,6 +129,26 @@ describe("install", () => {
         });
       },
     );
+
+    await withTempEnv(
+      ["i", "--save-optional", "@std/encoding@0.216.0"],
+      async (getPkgJson, dir) => {
+        assert.ok(
+          await isFile(path.join(dir, ".yarnrc.yml")),
+          "yarnrc file not created",
+        );
+        const pkgJson = await getPkgJson();
+        assert.deepEqual(pkgJson.optionalDependencies, {
+          "@std/encoding": "npm:@jsr/std__encoding@0.216.0",
+        });
+      },
+      {
+        prepare: async (dir) => {
+          await enableYarnBerry(dir);
+          await fs.promises.writeFile(path.join(dir, "yarn.lock"), "");
+        },
```

Comment:
It's a bit weird to not be able to read the code from top to bottom. With the pattern used in https://github.com/jsr-io/jsr-npm/blob/0209631bb79363e79d06932b204b696f3f214612/test/commands.test.ts#L346-L360 we can preserve that.

```ts
await runInTempDir(async (dir) => {
  // Preparation code here...
  
  // Call JSR
  await runJsr(args, dir);
});
```

Reply:
have refactored as so and removed the changes to `withTempDir`